docs/design2.md

```diff
@@ -0,0 +1,109 @@
+
+| Update Latency | History Audit    | Key Lookup |
+|----------------|------------------|------------|
+| hr             | 45Kb/day         | ~2Kb       |
+| min            | 2Mb/day          | ~2Kb       |
+| s              | 150Mb/day        | ~2Kb       |
+
+
+Key Transparency 2.0 is a new set of algorithms that removes this fundamental trade-off by sharing
+the task of auditing between pairs of clients. As a result, all configurations have efficiently
+auditable data structures that no not require the presence of 3rd party auditors.
+
+We expect that many applications will want to prioritize lower latency and
+reliability by using a *trust, then verify* mode whereby clients receive key
+updates that they can later verify as being part of the global, consistent
+state.  However, this does not preclude individual clients that have a lower
+tolerance for risk from waiting for full, globally consistent verification
+before using fresh public keys from their peers.
+
+| Update Latency | History Audit    | Key Lookup |
+|----------------|------------------|------------|
+| hr             | 18Kb O(1)        | ~2Kb       |
+| min            | 30Kb O(1)        | ~2Kb       |
+| s              | 40Kb O(1)        | ~2Kb       |
+
+
+## Data Structures
+
+### Gossip Network
+The job of the gossip network is to ensure that there is a single, globally consistent, lineage of log roots.
+To keep ![n^2](https://render.githubusercontent.com/render/math?math=n%5E2) communication costs low, and to prevent sybil attacks, clients use a small ~20 set of gossip nodes.
+
+Each gossip node fetches the latest signed log root (SLR) and verifies
+consistency with all previously seen roots with a single consistency proof.
+After verifying, the gossip node signs the log root.
+
+Gossip nodes offer the following APIs:
+
+1. Get Signed Root
+
+### Log of Map Roots
+
+The log server collects signatures from the gossip nodes to produce a Signed Log Root that has a quorum of gossip signatures.
+
+The log server contains a list of map roots representing sequential snapshots of a key-value directory.
+
+Log servers offer the following APIs:
+
+1. Latest Signed Log Root
+1. Get consistency proof between log size a and b.
+1. Get log item i with inclusion proof to log size a.
+
+### Map Root Snapshots
+Each Map Root represents a snapshot of a key-value dictionary.
+The map is implemented as a sparse Merkle tree of fixed depth.
+
+* Indexes in the map are randomized and privacy protected via a Verifiable Random Function.
+* Values in the map represent the full history of values that have ever been stored at this index. This is accomplished by storing the Merkle root of a mini-log of these values.
+
+The map offers the following APIs:
+
+1. Get map value at index j with inclusion proof to snapshot a.
+
+### Value History Log
+These mini logs store not just the latest value, but also store every previous value in order.
```

Comment:
Please use "mini-log" or "mini log" consistently.

Reply:
Thanks for catching that